internal/taskupdate/manager.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2025 The A2A Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package taskupdate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/a2aproject/a2a-go/a2a"
+)
+
+type Saver interface {
+	Save(ctx context.Context, task *a2a.Task) error
+}
+
+type Manager struct {
```

Comment:
nit: Even though it's internal package, would be nice if we try to document exported symbols.